src/tests/java/htsjdk/samtools/sra/AbstractSRATest.java

```diff
@@ -0,0 +1,56 @@
+package htsjdk.samtools.sra;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.NoSuchElementException;
+
+@Test(groups = "SRA")
+public abstract class AbstractSRATest {
+
+    @BeforeMethod
+    public final void assertSRAIsSupported(){
+        if(!SRAAccession.isSupported()){
+            throw new SkipException("Skipping SRA Test because SRA native code is unavailable.");
+        }
+    }
+
+    protected static void checkAlignedUnalignedCountsByIterator(SAMRecordIterator samRecordIterator,
+                                                              int numberAlignments, int numberUnalignments) {
+        int countAlignments = 0, countUnalignments = 0;
+        while (
+                true) {
+            boolean hasRecord = samRecordIterator.hasNext();
+            SAMRecord record = null;
+            try {
+                record = samRecordIterator.next();
+                Assert.assertTrue(hasRecord); // exception is not thrown if we came to this point
+            } catch (NoSuchElementException e) {
+                Assert.assertFalse(hasRecord);
+            }
+
+            Assert.assertEquals(hasRecord, record != null);
+
+            if (record == null) {
+                break;
+            }
+
+            if (record.getReadUnmappedFlag()) {
+                countUnalignments++;
+            } else {
+                countAlignments++;
+            }
+        }
+
+        if (numberAlignments != -1) {
```

Comment:
Could you add doc stating that if this is `-1` that the values wont be checked, only `hasRecord`?